src/Miso/Property.hs

```diff
@@ -0,0 +1,61 @@
+-----------------------------------------------------------------------------
+{-# LANGUAGE OverloadedStrings #-}
+-----------------------------------------------------------------------------
+-- |
+-- Module      :  Miso.Property
+-- Copyright   :  (C) 2016-2025 David M. Johnson
+-- License     :  BSD3-style (see the file LICENSE)
+-- Maintainer  :  David M. Johnson <code@dmj.io>
+-- Stability   :  experimental
+-- Portability :  non-portable
+--
+-- Construct custom properties on DOM elements
+--
+-- > div_ [ prop "id" "foo" ] [ ]
+--
+----------------------------------------------------------------------------
+module Miso.Property
+  ( -- *** Smart constructors
+    textProp
+  , stringProp
+  , boolProp
+  , intProp
+  , integerProp
+  , doubleProp
+  , prop
+  ) where
+-----------------------------------------------------------------------------
+import           Data.Aeson (ToJSON(..))
+-----------------------------------------------------------------------------
+import           Miso.Types
+import           Miso.String (MisoString)
+-----------------------------------------------------------------------------
+-- | @prop k v@ is an attribute that will set the attribute @k@ of the DOM 
+-- node associated with the vnode to @v@.
+prop :: ToJSON a => MisoString -> a -> Attribute action
+prop k v = Property k (toJSON v)
+-----------------------------------------------------------------------------
+-- | Set field to `Bool` value
+boolProp :: MisoString -> Bool -> Attribute action
+boolProp = prop
+-----------------------------------------------------------------------------
+-- | Set field to `String` value
+stringProp ::  MisoString -> String -> Attribute action
+stringProp = prop
+-----------------------------------------------------------------------------
+-- | Set field to `Text` value
+textProp ::  MisoString -> MisoString -> Attribute action
```

Comment:
```suggestion
textProp ::  MisoString -> Text -> Attribute action
```

Reply:
in that case just update the Haddock not to mention `Text` :slightly_smiling_face:

Reply:
MisoString becomes Text on GHC and stays JSString on WASM